google/cloud/forseti/scanner/scanner.py

```diff
@@ -14,12 +14,70 @@
 """GCP Resource scanner."""
 
 from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.common.util.index_state import IndexState
 from google.cloud.forseti.scanner import scanner_builder
 from google.cloud.forseti.services.scanner import dao as scanner_dao
 
 LOGGER = logger.get_logger(__name__)
 
 
+def init_scanner_index(service_config):
+    """Initialize the 'scanner_index' table.
+
+    Make sure we have a 'scanner_index' row for the current scanner run.
+
+    Args:
+        service_config (ServiceConfig): Forseti 2.0 service configs.
+
+    Returns:
+        str: the id of the 'scanner_index' db row
+    """
+    with service_config.scoped_session() as session:
+        scanner_index = scanner_dao.ScannerIndex.create()
+        scanner_index.scanner_status = IndexState.RUNNING
+        session.add(scanner_index)
+        session.flush()
+        return scanner_index.id
+
+
+def _error_message(failed):
+    """Construct error message for failed scanners.
+
+    Args:
+        failed (list): names of scanners that failed
+
+    Returns:
+        str: error message detailing the scanners that failed
+    """
+    return 'Scanner(s) with errors: %s' % ', '.join(failed)
```

Comment:
We should add bit more message to explain what the error might mean: `Scanner(s) had errors and were unable to finish scaning for violations.`